### PR TITLE
issue #128: use reflection optionally to load target classes of intents.

### PR DESCRIPTION
**Pull request by stephanenicolas**
Solves #128 

this PR introduces a new annotation processor option for the henson processor: 
`dart.henson.useReflection=[true|false]`
which defaults to false.

TODO:
* [ ] add a wiki entry about the option

### CODE REVIEW

File: henson-processor/src/main/java/com/f2prateek/dart/henson/processor/IntentBuilderGenerator.java [hunk 1]
```diff
@@ -78,8 +80,13 @@ private void emitFields(TypeSpec.Builder intentBuilderTypeBuilder) {
   private void emitConstructor(TypeSpec.Builder intentBuilderTypeBuilder) {
     MethodSpec.Builder constructorBuilder = MethodSpec.constructorBuilder()
         .addModifiers(Modifier.PUBLIC)
-        .addParameter(ClassName.get("android.content", "Context"), "context")
-        .addStatement("intent = new Intent(context, $L)", target.className + ".class");
+        .addParameter(ClassName.get("android.content", "Context"), "context");
+    if(usesReflection) {
+        constructorBuilder.addStatement("try { intent = new Intent(context, Class.forName($S)); } catch(Exception ex)"
+                + " { throw new RuntimeException(ex); }", target.getFqcn());
```
Comment (dlemures) [inline henson-processor/src/main/java/com/f2prateek/dart/henson/processor/IntentBuilderGenerator.java]: Maybe use a proper error msg

**Comment by dlemures** [comment]
Just 1 comment and merge when green

File: henson-processor/src/main/java/com/f2prateek/dart/henson/processor/HensonExtraProcessor.java [hunk 2]
```diff
@@ -98,7 +101,7 @@
         Writer writer = null;
         try {
           IntentBuilderGenerator intentBuilderGenerator =
-              new IntentBuilderGenerator(injectionTarget);
+              new IntentBuilderGenerator(injectionTarget, usesReflection);
```
Comment (Macarse) [inline henson-processor/src/main/java/com/f2prateek/dart/henson/processor/HensonExtraProcessor.java]: `useReflection`

File: henson-processor/src/main/java/com/f2prateek/dart/henson/processor/IntentBuilderGenerator.java (hunk 1 above)
Comment (stephanenicolas) [inline henson-processor/src/main/java/com/f2prateek/dart/henson/processor/IntentBuilderGenerator.java]: This exception cannot happen.

**Comment by stephanenicolas** [comment]
@f2prateek we wanna merge this and release. We need it for an advanced problem here @ Groupon related to the size of our first dex ... Henson class was adding all activities in the first dex, we want to avoid that.

**Comment by f2prateek** [comment]
Sure sgtm, feel free to merge and release

**Comment by dlemures** [comment]
👍

**Comment by stephanenicolas** [comment]
Thx, @f2prateek ;)
Thx @dlemures & @Macarse for the review.

**Pull request merged**